File: src/main/java/com/wikia/tibia/repositories/KeyRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.KeyGateway;
import com.wikia.tibia.objects.Key;

public class KeyRepository extends WikiObjectRepository {

    public KeyRepository() {
        super(Key.class, new KeyGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/KeyRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Key

class KeyRepository : WikiObjectRepository<Key>(
    wikiObjectClass = Key::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.KEYS)
)
